### ISSUE TITLE
setStatusBarHeight does not exists on type

### ISSUE THREAD

**Issue by BillyFigueroa** [react-native-community/react-native-safe-area-view#91]
# Bug

I am trying to use setStatusBarHeight on SafeAreaView and I am getting the following TypeScript error

## Environment info
```json
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@expo/samples": "~3.0.3",
    "@expo/vector-icons": "^10.0.3",
    "@react-navigation/web": "^1.0.0-alpha.9",
    "axios": "^0.19.0",
    "expo": "^34.0.1",
    "expo-asset": "^6.0.0",
    "expo-constants": "6.0.0",
    "expo-font": "~6.0.0",
    "expo-web-browser": "6.0.0",
    "jwt-decode": "^2.2.0",
    "moment": "^2.24.0",
    "native-base": "^2.13.5",
    "react": "16.8.3",
    "react-dom": "^16.8.6",
    "react-moment": "^0.9.2",
    "react-native": "https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz",
    "react-native-calendars": "^1.209.0",
    "react-native-circular-progress": "^1.3.0",
    "react-native-dropdownalert": "^4.1.1",
    "react-native-gesture-handler": "~1.3.0",
    "react-native-svg": "^9.7.0",
    "react-native-swipeout": "^2.3.6",
    "react-native-web": "^0.11.4",
    "react-navigation": "^3.11.0",
    "react-redux": "^7.1.0",
    "redux": "^4.0.4",
    "redux-promise-middleware": "^6.1.1",
    "redux-thunk": "^2.3.0",
    "validate.js": "^0.13.1"
  },
  "devDependencies": {
    "@types/react": "^16.8.23",
    "@types/react-native": "^0.57.65",
    "babel-preset-expo": "^6.0.0",
    "redux-logger": "^3.0.6",
    "typescript": "^3.4.5"
  },
  "private": true
}

```

React native info output:

```bash
Property 'setStatusBarHeight' does not exist on type 'ComponentClass<SafeAreaViewProps, any>'.
```

![image1](https://user-images.githubusercontent.com/1371105/64446375-9f767580-d0a6-11e9-9fdc-2337170a66ed.png)

This is an image on iPhone but I only did that because I can easily transfer screenshots from my iPhone to iMac computer. This is the same error I see on android (s9). however, It is a problem on both platforms

Library version: "0.14.7"

## Steps To Reproduce

I am trying to use the following logic

```js
{ Platform.OS === 'android' && SafeView.setStatusBarHeight(0) }
```

I expect that this would not cause an error in TypeScript and I would be able to set the height

**Comment by BillyFigueroa** [comment]
FIY I tried to add some quick changes to the type definition file in the react-nativ-safe-area-view folder and it did nothing. I have never tried to update a type file before but what I did was basically changed to this


```javascript
export interface SafeAreaViewProps extends ViewProperties {
  forceInset?: {
    top?: SafeAreaViewForceInsetValue;
    bottom?: SafeAreaViewForceInsetValue;
    left?: SafeAreaViewForceInsetValue;
    right?: SafeAreaViewForceInsetValue;
    horizontal?: SafeAreaViewForceInsetValue;
    vertical?: SafeAreaViewForceInsetValue;
  };
  setStatusBarHeight: (heght: number) => void;
}

```

and this

```javascript
export interface SafeAreaViewProps extends ViewProperties {
  forceInset?: {
    top?: SafeAreaViewForceInsetValue;
    bottom?: SafeAreaViewForceInsetValue;
    left?: SafeAreaViewForceInsetValue;
    right?: SafeAreaViewForceInsetValue;
    horizontal?: SafeAreaViewForceInsetValue;
    vertical?: SafeAreaViewForceInsetValue;
  };
}

export const getStatusBarHeight: (isLandscape?: boolean) => number;
export const setStatusBarHeight: (height: number) => void;

```

and neither of these removed the typescript warning/ error. Is there a proper way to do this that I am not following?

**Comment by chinesedfan** [comment]
@BillyFigueroa Please try #92. The key point is that those 2 methods are static methods.

**Comment by BillyFigueroa** [comment]
@chinesedfan I just copied your code into the local package I have (not sure how to pull down your code) and it now shows up in auto complete for intellisense but there is still an error.

`
undefined is not an object (evaluating '_reactNativeSafeAreaView.SafeAreaView.setStatusBarHeight')
`

**Comment by chinesedfan** [comment]
@BillyFigueroa Looks like `SafeAreaView` is not imported as expected, because the error says `xx is not an object`, instead of `xx is not a function` (means `setStatusBarHeight` is not found).

You can check my runnable [project](https://github.com/gitpoint/git-point/blob/master/App.js#L37). Or give more details about your codes.

**Comment by BillyFigueroa** [comment]
@chinesedfan thanks for continuing to help

Here is basically what I was doing

```js
...
import { SafeAreaView } from 'react-native-safe-area-view';

if (Platform.OS === 'android') {
  SafeAreaView.setStatusBarHeight(0);
}
...
```

So here is what I noticed...

1) You are importing SafeAreaView from `react-navigation` not from `react-native-safe-area-view`
2) If I just throw in your code to modify the SafeAreaView export from react-navigation's index.d.ts file than it works, but when I try to add the same change to `react-native-safe-area-view` than it does not work and that is when I get that `undefined is not an object` error

What is going on? This proves the SafeAreaView from react-navigation works if I make your change but am I misunderstanding the `react-native-safe-area-view` package?. Why are you importing from `react-navigation`?

**Comment by chinesedfan** [comment]
@BillyFigueroa Ah, I see. You might be the first man to use this package in TS. 😂

Its js implementation only has the default export. So change like following will resolve your problem.

```
- import { SafeAreaView } from 'react-native-safe-area-view';
+ import SafeAreaView from 'react-native-safe-area-view';
```

**Comment by BillyFigueroa** [comment]
@chinesedfan 🤦🏽

Stupid programmer err...

And yea I switched to using TypeScript. I had a project running react navigation 1.5 I think it was and tried to upgrade to use react navigation 3 and it was going horrible lol so I created a new project in typescript and set up react navigation and then copied my project component by component slowly.

That went easier than trying to figure out how to make it work with older versions of packages like redux and others

Anyhow, thanks for the help. Your fix works when it is imported properly lol
